### PR TITLE
Fix tasks lockup

### PR DESCRIPTION
**Pull request by drmarkwslater**
This should fix the Tasks lockup seen in the testing. However, I can't recreate this locally so I guess we'll have to wait and see...

### CODE REVIEW

File: python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py [hunk 1]
```diff
@@ -44,6 +53,7 @@ def test_a_TaskCreation(self):
     def test_b_TaskPersistency(self):
         """Check the task is persisted properly"""
         from Ganga.GPIDev.Base.Proxy import isType
+        from Ganga.GPI import tasks
```
Comment (milliams) [inline python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py]: This line duplicates the one below.

File: python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py [hunk 2]
```diff
@@ -85,3 +95,30 @@ def test_d_TaskStillRunning(self):
...
+            
+        assert len(tasks) == self._numTasks + 1
+
+    def test_f_CheckAccessToManyTasks(self):
+        """Access the Tasks while monitoring"""
+        from Ganga.GPI import tasks, CoreTask, CoreTransform, Executable, GenericSplitter
+        from Ganga.GPIDev.Base.Proxy import isType
+
+        for _ in range(0, self._numTasks):
+            for t in tasks:
```
Comment (milliams) [inline python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py]: Why does this loop over the number of tasks and then again over each task?

File: python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py (hunk 1 above)
Comment (drmarkwslater) [inline python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py]: Fixed!

File: python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py (hunk 2 above)
Comment (drmarkwslater) [inline python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py]: Just to keep hammering the access to Tasks while the monitoring is loading. I guess I can just replace this with a random number but I wanted it to scale with the number of Tasks that are being generated.

As I say though, I can't get this to fail locally but should fail make the failure more obvious on the testing machines.

Comment (milliams) [inline python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py]: Ok, that makes sense but perhaps a comment to that effect would be useful for 'future us'.

Comment (drmarkwslater) [inline python/Ganga/new_tests/GPI/Tasks/TestCoreTasks.py]: Added!

**Comment by drmarkwslater** [comment]
Any objections to a merge after the tests complete?

**Comment by milliams** [comment]
:+1:

**Comment by rob-c** [comment]
Looks good to me :+1:

**Pull request merged**